Field flow: Keep URL params matching HTML attributes

EricDunsworth opened the pull request:
When field flow's redirect (``redir``) action is used, submitting causes the plugin to "transform" the currently-selected dropdown ``option``'s URL parameters into hidden ``input`` elements. The ``input``s are created by passing "raw" HTML strings to the jQuery object.

That setup used to play nicely with URL parameters whose keys corresponded to the names of HTML attributes (e.g. ``lang=anything``). But it stopped working when wet-boew/wet-boew#9210 introduced DOMPurify into WET's jQuery 2.x implementation. Why? Because DOMPurify's ``sanitize()`` method filters-out ``name="[any HTML attribute name]"`` to prevent potential DOM clobbering attacks (see cure53/DOMPurify#980). End result is that jQuery ultimately returns ``name``-less ``input``s to the plugin, which in turn causes affected parameters to go missing.

This fixes it by using "pure" JavaScript (instead of jQuery) to create the ``input``s. Also adds a query string example to the redirection demo (with a mix of key naming schemes).

Fixes #2406.

EricDunsworth commented on components/wb-fieldflow/fieldflow.js:
```diff
@@ -1207,9 +1207,14 @@ $document.on( "submit", selectorForm + " form", function( event ) {
+					hdnField = document.createElement( "input" );
+					hdnField.setAttribute( "type", "hidden" );
+					hdnField.setAttribute( "name", cacheName );
+					hdnField.setAttribute( "value", wb.escapeAttribute( cacheParam ) );
```
Should I get rid of the call to [``wb.escapeAttribute()``](https://github.com/wet-boew/wet-boew/blob/e01bc308c1cc123dbe7948cd8dccc14829ed8e70/src/core/helpers.js#L1313-L1315) at this spot?

I don't think ``setAttribute()`` technically has a need for it, but figured it can't hurt to ask before going through with it.

EricDunsworth replied:
Sorry - must've overlooked this :(.

In any case, I ended up implementing @duboisp's suggestion to set the attribute properties directly instead of using ``setAttribute()`` later on. Given that, I think it might be a safer bet to keep ``wb.escapeAttribute()`` in play.

I won't bother sending another PR to remove it.

Garneauma commented:
Pre-approved upon successful review.

duboisp commented on components/wb-fieldflow/fieldflow.js:
```diff
@@ -1207,9 +1207,14 @@ $document.on( "submit", selectorForm + " form", function( event ) {
-					wbRegisteredHidden.push( $hdnField.get( 0 ) );
+
+					hdnField = document.createElement( "input" );
+					hdnField.setAttribute( "type", "hidden" );
+					hdnField.setAttribute( "name", cacheName );
+					hdnField.setAttribute( "value", wb.escapeAttribute( cacheParam ) );
```
```suggestion
hdnField.type = "hidden";
hdnField.name = cacheName;
hdnField.value = wb.escapeAttribute( cacheParam );
```

duboisp reviewed (changes requested):
To review and fully test.

This is a **patch**.

EricDunsworth commented:
FWIW I retested locally with @duboisp's suggested change and it still worked fine 💯.

EricDunsworth commented:
Rebased, updated modified dates and force-pushed.

This wasn't running into any merge conflicts, but since it hadn't been tested by any project maintainers yet... I figured it'd make sense to rebase to make it node 20-friendly.

Garneauma reviewed (approved):
Tested locally. Works as intended.

MatthieuBonenfant-GOC commented:
Thank you all for your contribution!
